auction-app-frontend/src/components/LocationForm/LocationForm.jsx

```diff
@@ -0,0 +1,145 @@
+import { useEffect, useState } from "react";
+import { set, useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+
+import { FormContainer, LoadingComponent, ErrorComponent } from "src/components";
+
+import { locationShippingFormFields, cardInformationFields } from "src/forms/fields";
+import { BUTTON_LABELS, ROUTE_PATHS } from "src/constants";
+import { getPaymentInfoByUser } from "src/services";
+import { useUser } from "src/store/UserContext";
+
+import "./style.scss";
+
+const LocationForm = ({ formData, setFormData, handleFinalSubmit }) => {
+    const navigate = useNavigate();
+
+    const { userId, email } = useUser(); 
+
+    const [originalPaymentInfo, setOriginalPaymentInfo] = useState(formData);
+    const [showCardInfo, setShowCardInfo] = useState(false);
+    const [paymentInfo, setPaymentInfo] = useState({});
+    const [error, setError] = useState(null);
+    const [loading, setLoading] = useState(false);
+
+    const methods = useForm({
+        mode: "onBlur", 
+        defaultValues: formData
+    });
+
+    const fetchPaymentInfo = () => {
+        if (!userId) return;
+    
+        setLoading(true);
+    
+        getPaymentInfoByUser(userId)
+            .then((response) => {
+                const { expirationDate, ...rest } = response;
+                const [year, month, day] = expirationDate.split('-');
+        
+                const expirationMonth = month;
+                const expirationYear = year.substring(2);
+        
+                const updatedPaymentInfo = {
+                    ...rest,
+                    expirationMonth,
+                    expirationYear,
+                    email
+                };
+    
+                setPaymentInfo(updatedPaymentInfo);
+                setOriginalPaymentInfo(updatedPaymentInfo); // for later use
```

Comment:
If you want to leave comment to be more descriptive, than your comment has to say user something valuble.
We set originalPaymentInfo for later use, but why, just give it some context